XLForm/XL/Helpers/UIView+XLFormAdditions.m

```diff
@@ -27,47 +27,60 @@
 
 @implementation UIView (XLFormAdditions)
 
-+ (id)autolayoutView
++ (instancetype)autolayoutView
 {
-    UIView *view = [self new];
+    __kindof UIView *view = [self new];
     view.translatesAutoresizingMaskIntoConstraints = NO;
+    
     return view;
 }
 
--(NSLayoutConstraint *)layoutConstraintSameHeightOf:(UIView *)view
+- (NSLayoutConstraint *)layoutConstraintSameHeightOf:(UIView *)view
 {
-    return [NSLayoutConstraint constraintWithItem:self attribute:NSLayoutAttributeHeight relatedBy:NSLayoutRelationEqual toItem:view attribute:NSLayoutAttributeHeight multiplier:1.0f constant:0.0f];
+    return [NSLayoutConstraint constraintWithItem:self
+                                        attribute:NSLayoutAttributeHeight
+                                        relatedBy:NSLayoutRelationEqual
+                                           toItem:view
+                                        attribute:NSLayoutAttributeHeight
+                                       multiplier:1.0
+                                         constant:0.0];
 }
 
 - (UIView *)findFirstResponder
 {
+    UIView *firstResponder = nil;
     if (self.isFirstResponder) {
-        return self;
+        firstResponder = self;
```

Comment:
I do not understand this change. If the view is the first responder then the search finished and `self` should be returned, right?

Reply:
to make one exit from function

Reply:
Okay, but we should not loop through the `subviews` if `self.isFirstResponder`

Reply:
fixed in 509ea4d